src/NetworkUtil.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace Castor;
+
+use Castor\Exception\WaitForInvalidCallbackCheckException;
+use Castor\Exception\WaitForTimeoutReachedException;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+use function Symfony\Component\String\u;
+
+class NetworkUtil
+{
+    /**
+     * @param callable $callback function(): bool|null (return null to break the loop, true if is OK, false otherwise)
+     *
+     * @throws WaitForTimeoutReachedException
+     */
+    public static function wait_for(
+        callable $callback,
+        int $timeout = 10,
+        string $name = null,
+        bool $throw = false,
+        bool $quiet = false,
+        int $intervalMs = 100,
+        string $message = null,
+        string $successMessage = null,
+    ): bool {
+        if ($name) {
+            if ($message) {
+                $msg = sprintf($message, $name);
+            } else {
+                $msg = "Waiting for {$name} to be available...";
+            }
+            if ($successMessage) {
+                $successMessage = sprintf($successMessage, $name);
+            } else {
+                $successMessage = " <fg=green> OK {$name} is available !</>";
+            }
+        } else {
+            $msg = $message ?? 'Waiting for callback to be available...';
+            $successMessage ??= ' <fg=green> OK callback is available !</>';
+        }
+
+        if (false === $quiet) {
+            io()->write($msg);
+        }
+
+        $start = time();
+        $end = $start + $timeout;
+
+        $elapsed = 0;
+        while (time() < $end) {
+            $elapsed += $intervalMs;
+            if (false === $quiet && 0 === $elapsed % 1000) {
```

Comment:
The reason I've done this is that if a service, for example in my case Keycloak, launches its container, and the time it takes to boot and access the health check can last a minimum of four seconds generally. This would display 40 dots to 60 dots on average. I think this keeps things visually consistent?
That's why I preferred to limit the number of dots to one per second

Reply:
Hmm, got it, You don't recompute `$elapsed`. This is wrong :) I'll fix that later, don't worry